packages/react-vapor/src/components/limit/Limit.tsx

```diff
@@ -0,0 +1,210 @@
+import classNames from 'classnames';
+import * as VaporSVG from 'coveo-styleguide';
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {InputConnected, Svg} from 'react-vapor';
+import {IReactVaporState} from '../../ReactVapor';
+import {IDispatch} from '../../utils/ReduxUtils';
+import {changeInputValue} from '../input/InputActions';
+import {InputSelectors} from '../input/InputSelectors';
+
+export interface LimitOwnProps {
+    id?: string;
+    title: string;
+    limitLabel?: string;
+    usage?: number;
+    limit?: number;
+    isLimitEditable?: boolean;
+    isLimitTheGoalToReach?: boolean;
+    isHistoryIncluded?: boolean;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: LimitOwnProps) => {
+    const ownLimit: string = ownProps.limit.toString() || Limit.defaultProps.limit.toString();
+    const currentLimit: string = InputSelectors.getValue(state, {id: ownProps.id}) || ownLimit;
+    return {
+        currentLimit,
+    };
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: LimitOwnProps) => ({
+    onChangeLimit: (limitValue: string) => {
+        dispatch(changeInputValue(ownProps.id, limitValue, true));
+    },
+});
+
+export interface LimitProps
+    extends LimitOwnProps,
+        ReturnType<typeof mapDispatchToProps>,
+        ReturnType<typeof mapStateToProps> {}
+
+export const LimitDisconnect: React.FunctionComponent<LimitProps> = ({
+    id,
+    title,
+    limitLabel,
+    usage,
+    limit,
+    currentLimit,
+    isLimitEditable,
+    isLimitTheGoalToReach,
+    isHistoryIncluded,
+    onChangeLimit: onChangeLimit,
+}) => {
+    return (
+        <div className="limit-box mb2">
+            <div className="limit-box-main p2 pb1">
+                <HeaderDivision limitTitle={title} isHistory={isHistoryIncluded} />
+                <ContentDivision
+                    id={id}
+                    usageValue={usage}
+                    limitValue={limit}
+                    isEditable={isLimitEditable}
+                    limitTitle={limitLabel}
+                    onChangeLimit={onChangeLimit}
+                />
+            </div>
+            <ProgressBar usageValue={usage} isLimitTheGoal={isLimitTheGoalToReach} limitValue={+currentLimit} />
+        </div>
+    );
+};
+
+const HeaderDivision = ({limitTitle, isHistory}: {limitTitle: string; isHistory: boolean}) => {
+    return (
+        <div className="flex space-between">
+            <label className="form-control-label"> {limitTitle}</label>
+            <HistoryIcon isHistory={isHistory} />
+        </div>
+    );
+};
+
+const HistoryIcon = ({isHistory}: {isHistory: boolean}) => {
+    return (
+        isHistory && (
+            <span className="icon mod-lg limit-history-button">
+                <Svg svgName={VaporSVG.svg.menuAnalytics.name} className="fill-medium-blue" />
+            </span>
+        )
+    );
+};
+
+const ContentDivision = ({
```

Comment:
this should be a React.FunctionComponent.  `const ContentDivision: React.FunctionComponent<TYPE OF THE PROPS> = (PROPS) => {}`